### PR TITLE
Fix BF16 CPX code

### PR DESCRIPTION
**Pull request by deeptiag1**
Tested for following:
 cpx BF16  , cpx SP  ,cpx DP  , cpx BF16F32  , clx BF16, clx BF16F32

### CODE REVIEW

File: src/generator_gemm_avx512_microkernel.c [hunk 1]
```diff
@@ -2551,8 +2551,7 @@ LIBXSMM_API_INTERN void libxsmm_generator_gemm_avx512_microkernel_fsdbcst( libxs
             LIBXSMM_HANDLE_ERROR( io_generated_code, LIBXSMM_ERR_ARCH_PREC );
             return;
           }
-        } else if (LIBXSMM_GEMM_PRECISION_BF16 == LIBXSMM_GETENUM_INP( i_xgemm_desc->datatype ) &&  io_generated_code->arch < LIBXSMM_X86_AVX512_CPX
-                   && io_generated_code->arch != LIBXSMM_X86_AVX512_VL256_CPX ) {
+        } else if (LIBXSMM_GEMM_PRECISION_BF16 == LIBXSMM_GETENUM_INP( i_xgemm_desc->datatype )) {
```
Comment (alheinecke) [inline src/generator_gemm_avx512_microkernel.c:2554]: this change looks fishy... what is the original code doing?

**Review by alheinecke — CHANGES_REQUESTED** [review]
one change looks fishy and can you please exclude all changes to mateltwise/matequation from this PR?

File: src/generator_gemm_avx512_microkernel.c (hunk 1 above)
Comment (deeptiag1) [reply src/generator_gemm_avx512_microkernel.c:2554]: This change is reverting the  code state  to original code. 
Original Code is doing the same thing https://github.com/hfp/libxsmm/blob/d46a8f2f0104d82fd37d6e157ebab0e39be1b7fb/src/generator_gemm_avx512_microkernel.c#L1442.

**Comment by alheinecke** [comment]
OK, then after removal of the eltwise stuff, this can be merged

**Comment by deeptiag1** [comment]
Reverted eltwise changes

File: src/generator_mateltwise_misc_avx_avx512.c [hunk 2]
```diff
@@ -97,8 +97,9 @@ void libxsmm_generator_mn_code_block_replicate_col_var_avx_avx512( libxsmm_gener
       }
 
       if (downconvert_input_f32bf16 > 0) {
-        if (io_generated_code->arch < LIBXSMM_X86_AVX512_CPX) {
-          libxsmm_generator_vcvtneps2bf16_avx512_preppedstack( io_generated_code, 'z',
+        if  (io_generated_code->arch < LIBXSMM_X86_AVX512_CPX) {
```
Comment (alheinecke) [inline src/generator_mateltwise_misc_avx_avx512.c:100]: can you please revert this file as well?

**Review by alheinecke — CHANGES_REQUESTED** [review]
very minor change

File: src/generator_mateltwise_misc_avx_avx512.c (hunk 2 above)
Comment (deeptiag1) [reply src/generator_mateltwise_misc_avx_avx512.c]: Done!

**Review by alheinecke — APPROVED** [review]

**Pull request merged**